### PR TITLE
docs on auth0 config and rules

### PR DESCRIPTION
**Pull request by scottyhq**
I just made a copy of all the rules we're currently using on the Auth0 account so that they are accessible in version control. Also some brief instructions on setup. Discussed further in https://github.com/pangeo-data/pangeo-cloud-federation/issues/611

### CODE REVIEW

File: auth0/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+This folder contains instructions on setting up Auth0 for pangeo user management. Auth0 'rules' are javascript functions that run when a user logs in to a pangeo hub. Currently we use them to check the GitHub Organizations and teams that a user belongs to, and grant access accordingly. The order that rules run in matters. Environment variables must be set under both 'Rules' and per 'Application' (under 'advanced settings')
+
+## Setup
+Comments in the rules describe necessary configuration. In brief, you need a 'Regular Web Application' per hub, and a single 'Machine to Machine' auth0 management app that rules use to modify user metadata.
+
+## Environment Variables
+Under 'Rules' in the Auth0 dashboard you'll need:
+AUTH0_DOMAIN_NAME=xxx
+MANAGEMENT_CLIENT_ID=xxx
+MANAGEMENT_CLIENT_SECRET=xxx
```
Comment (jhamman) [inline auth0/README.md]: can you add a bit more on where these come from?

Comment (scottyhq) [reply auth0/README.md]: just did a major revamp of README @jhamman - hopefully it's easier to follow start to finish

File: auth0/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+### Machine to Machine ('Management Client')
+You have to create an Management API Application with proper scope to allow Rules to modify Auth0 metadata. 
+
+- Go to Applications, create new application
+- Name: Auth0 Management API
+- Type: Machine to Machine App
+- Select API: Auth0 Management API
+- Go to Application settings, you'll find your ID and SECRET (to use as Rules Environment Variables)
```
Comment (jhamman) [inline auth0/README.md:36]: The one missing piece here is a list of scopes needed in this application: 

image

Above you write: "with proper scope to allow Rules to modify Auth0 metadata". Its not clear at first glance which scopes are needed though.

Comment (scottyhq) [reply auth0/README.md:36]: good catch. just added that

**Pull request merged**